### PR TITLE
Make package on top apply fix

### PR DESCRIPTION
**Pull request by AttilaTheFun**
Currently the package-on-top buildifier lint rule only shows a warning but is unable to offer a fix.
This PR addresses this issue by essentially cloning the behavior of the load-on-top rule, but placing the package statement after the load statements (as well as any docstrings or comments).
I updated the test to use checkFindingsAndFix as well and it passes.

### CODE REVIEW

File: warn/warn_cosmetic.go [hunk 1]
```diff
@@ -102,30 +102,75 @@ func sameOriginLoadWarning(f *build.File) []*LinterFinding {
...
-	seenRule := false
-	for _, stmt := range f.Stmt {
+	if f.Type == build.TypeWorkspace || f.Type == build.TypeModule {
+		// Not applicable to WORKSPACE or MODULE files
+		return nil
+	}
+
+	// Find the misplaced load statements
+	misplacedPackages := make(map[int]*build.CallExpr)
+	firstStmtIndex := -1 // index of the first seen non string, comment or lead statement
```
Comment (vladmos) [inline warn/warn_cosmetic.go:117]: typo: load

File: warn/warn_cosmetic.go [hunk 2]
```diff
@@ -102,30 +102,75 @@ func sameOriginLoadWarning(f *build.File) []*LinterFinding {
...
+			continue
+		}
+		misplacedPackages[i] = rule.Call
+	}
+
+	// Calculate a fix
+	if firstStmtIndex == -1 {
+		firstStmtIndex = 0
+	}
+	offset := len(misplacedPackages)
```
Comment (vladmos) [inline warn/warn_cosmetic.go:143]: Nit: add `if offset == 0 { return nil }` for better readability?

**Review by vladmos — APPROVED** [review]

**Comment by AttilaTheFun** [comment]
Thanks @vladmos ! I incorporated your suggestions and pushed the commit. I don't think I have permission to merge it, so if you could do so whenever you have a chance that would be great.

**Comment by vladmos** [comment]
I'll merge the pull request in a few days, I want to do it separately from other pull requests to make sure it doesn't break any internal projects.

**Pull request merged**